hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceManager.java

```diff
@@ -484,8 +484,10 @@ public void refreshSources(String peerId) throws IOException {
         LOG.info("Terminate replication source for " + toRemove.getPeerId());
         toRemove.terminate(terminateMessage);
       }
-      for (NavigableSet<String> walsByGroup : walsById.get(peerId).values()) {
-        walsByGroup.forEach(wal -> src.enqueueLog(new Path(this.logDir, wal)));
+      if (walsById.get(peerId) != null) {
```

Comment:
And is this possible? In the addSource method, we will always put a Map into the walsByGroup for the corresponding peerId. Notice that the code for branch-2 have been changed a lot from branch-1 so...